LibreNMS/OS/Routeros.php

```diff
@@ -85,22 +81,47 @@ public function discoverWirelessClients()
             '.1.3.6.1.4.1.14988.1.1.1.3.1.6.'
         );
     }
-
-    /**
+   /**
      * Discover wireless frequency.  This is in MHz. Type is frequency.
      * Returns an array of LibreNMS\Device\Sensor objects that have been discovered
      *
      * @return array Sensors
      */
     public function discoverWirelessFrequency()
+    {
+        $data = $this->fetchData();
+       
+        $sensors = array();
+        foreach ($data as $index => $entry) {
+            if ($entry['mtxrWlApFreq'] == null) {
+                return $this->discoverSensor(
+                    'frequency',
+                    'mtxrWl60GFreq',
+                    '.1.3.6.1.4.1.14988.1.1.1.8.1.6.'
+                );
+            } else {
+                    return $this->discoverSensor(
+                        'frequency',
+                        'mtxrWlApFreq',
+                        '.1.3.6.1.4.1.14988.1.1.1.3.1.7.'
+                    );
+            }
+        }
+    }
+    /**
+     * Discover wireless Rssi.  This is in Dbm. Type is Dbm.
+     * Returns an array of LibreNMS\Device\Sensor objects that have been discovered
+     *
+     * @return array Sensors
+     */
+    public function discoverWirelessRssi()
     {
         return $this->discoverSensor(
-            'frequency',
```

Comment:
You're getting close. I'll tidy up the code after but for now, why have you removed mtxrWlApFreq?